### ISSUE TITLE
Private pages not returned if logged in as site admin

### ISSUE THREAD

**Issue by eavonius** [wp-graphql/wp-graphql#2822]
### Description

I've got a page that's marked private which I can view in a typical wordpress theme if I'm logged in a site admin.

When I'm logged in as the same user (I use WPGraphQL CORS), I can still get the "viewer" in my graphql query and it shows the right email as me being logged into the same account, but the nodes list is empty (no data for the page).

### Steps to reproduce

1. Create a remix app (or otherwise) that can make a fetch call with the Apollo graphql npm package.
2. Set the app to run on port 3000 (or any port really).
1. Install the WPGraphQL CORS plugin.
3. Edit the WPGraphQL settings in the Wordpress Admin area and make the following changes:
   * Add http://localhost AND http://localhost:3000 (or whatever port your app runs on) to the "Extend "Access-Control-Allow-Origin” header" setting.
   * Make sure "Send site credentials." is checked.
   * Make sure "Enable login mutation" and "Enable logout mutation" are checked.
   * Set "Samesite cookie mode" to LAX.
   * Leave the cookie domain blank.
   * Save your changes.
4. Create a page in wordpress marked as private.
5. Add code to the app to call the login mutation and pass the cookie when creating your graphql client. I do something like this:
   ```typescript
   import { ApolloClient, createHttpLink, InMemoryCache } from "@apollo/client";
   import type {
	ApolloQueryResult,
	NormalizedCacheObject,
	OperationVariables,
	QueryOptions,
   } from "@apollo/client";
   
   export function createGraphqlServerClient(
   	request: Request
   ): ApolloClient<NormalizedCacheObject> {
	const cookie = request.headers.get("Cookie");

	const headers =
		cookie != null
			? {
					cookie,
			  }
			: undefined;

	const httpLink = createHttpLink({
		uri: `http://localhost/wp/graphql`,
		credentials: "include",
		headers,
	});

	return new ApolloClient({
		link: httpLink,
		cache: new InMemoryCache(),
		ssrMode: true,
	});
   }
   ```
6. Make a graphql call to retrieve that page and the viewer. e.g.
   ```graphql
   query {
        user: viewer {
           id
	   databaseId
	   username
	   email
	   firstName
	   lastName
        },
	pages(first: 1, where: { name: "your-private-page-slug", status: PRIVATE }) { #Status might be optional
	   nodes {
		id
		databaseId
		slug
		link
		title
          }
       }
   }
   ```
7. The nodes property that should return a page will be empty, while the viewer will be populated.

### WPGraphQL Version

1.14.3

### WordPress Version

6.2

### PHP Version

8.1.16

### Additional enviornment details

WPGraphQL CORS version 2.1

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for WPGraphQL.

- [ ] Yes
- [X] My issue is with compatibility with a specific WordPress plugin, and I have listed all my installed plugins (and version info) above.

**Comment by justlevine** [comment]
Thanks for taking the time to open this @eavonius !

I'm unclear from your description - is the issue with WPGraphQL core or specifically when using WPGraphQL CORS?
What happens if you run that query as an authenticated user in the WPGraphiQL IDE on backend (with CORS disabled)? Do you get the expected results?

**Comment by eavonius** [comment]
Just wanted to clarify (I was reading my comment and wondering if it was confusing). By "I get the same results in the IDE". I mean I get the same problem in the IDE with the nodes being empty. Hope that helps. Thanks!

**Comment by jasonbahl** [comment]
@eavonius can you provide steps to reproduce in WordPress + WPGraphQL alone?

Like: 

- login to WordPress as an admin users
- create a page
- set the page as private
- save the page
- query the page with xx query in GraphiQL as the same authenticated admin user that authored the page

What's the expected behavior? What's the actual behavior? What's the behavior in  the REST API? What's the behavior in non-headless WP?


Also, if you can reproduce in a clean environment with WordPress + WPGraphQL and no other plugins, that would be great to make sure it's a WPGraphQL bug an not something from another plugin. 

The more we can keep it to WPGraphQL and not the clients consuming WPGraphQL the better we can help.

**Comment by jasonbahl** [comment]
fwiw, I created a private page: 

![CleanShot 2023-06-02 at 09 28 40](https://github.com/wp-graphql/wp-graphql/assets/1260765/fc605839-6862-4076-bae7-eb6fd2c3a28c)

And was able to query it just fine as an authenticated user in the GraphiQL IDE:

![CleanShot 2023-06-02 at 09 28 14](https://github.com/wp-graphql/wp-graphql/assets/1260765/885a7776-6612-40b4-982b-26331d8a1f41)

Note, when using the GraphiQL IDE, you can toggle between public and authenticated using this toggle: 

![CleanShot 2023-06-02 at 09 29 56](https://github.com/wp-graphql/wp-graphql/assets/1260765/d5bf43ab-9c79-4697-9a47-d4b3601b36e6)

If you're seeing "viewer" information, then I would believe that you are indeed executing queries as an authenticated user, but just wanted to make sure.

**Comment by jasonbahl** [comment]
My hunch is that maybe the user you're authenticated as doesn't have proper capabilities to see private pages?

**Comment by eavonius** [comment]
Hey @jasonbahl I just realized I did NOT have the button checked in the IDE so I was logged in *facepalm*. I thought the icon indicated logged in. When I do, it does come back if the status is set to "Private". So that's good news...

The graphql query I use is for a remix route that matches based on slug name. It needs to pull back public pages or private pages for the current user based on the slug in the URL. Is there a way to specify multiple statuses in the where clause?
